engine/language_client_go/main.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+/*
+#cgo LDFLAGS: ./lib/libhello.dylib -ldl
+#include <stdlib.h>
+#include <stdbool.h>
+#include "./lib/baml.h"
+
+extern void trigger_callback(uint32_t, bool, const char *);
+*/
+import "C"
+
+import (
+	"fmt"
+	"unsafe"
+)
+
+type StreamResult[Partial any, Final any] struct {
```

Comment:
We should confirm if gophers apart from me are comfortable to get a stream of such an object

Reply:
What would the other option be?

Reply:
That's what I'd like to know as well. I recommend this way myself.

Reply:
I think regardless of channel vs callback this is definitely the 'right' data structure.

Curious though, and this may be my lack of knowledge w/ baml streaming. When would Partial and Final be a different type?